### ISSUE TITLE
Change default value for timeout

### ISSUE THREAD

**Issue by jeandut** [benchopt/benchopt#531]
Hi !
Currently the default timeout is 100 seconds. For inattentive users the abrupt stop of the computation is surprising and hard to debug. Maybe like 10 minutes (or 600 seconds) would be more sensible.
Also, not sure how to best implement that but it would be great if users could pass hours or minutes directly. One way to do that would be to overload how click CLI loads this argument and have a regexp parser converting the string to an int in seconds allowing syntax like:
```python
--timeout 3h #understood as 3 hours
--timeout 60 #understood as 60 seconds
--timeout 10m #understood as 10 minutes
```
Again thanks for the great tool !

**Comment by tomMoral** [comment]
Hello,

I actually like the default timeout to `100`as this avoids launching very large computations without thinking about how long it will take and this makes running a first example when you discover `benchopt ` or debugging runs fast to run. 

We could raise it to a larger timeout like `600` (which would still be compatible with the first goal) but then we loose the default "fast to run" property. Not sure what others think on this one? @agramfort @mathurinm ?

For the "hard to debug" part, I am not sure why you say it is surprising? The output is supposed to clearly state that the solver reached a timeout, showing a yellow `done (timeout)` . Is this not explicit when you run this (*e.g.* because we should display the timeout value) or did it not report this properly?

For passing the timeout with `m`, `h`, I am all in for this (this is tiresome to do the conversion). Maybe we can make use `pd.to_timedelta(x).total_seconds()` which should make the right conversion I think, except for integer which would need to be special cased.

**Comment by jeandut** [comment]
Hi !
Thanks for the answer maybe the term "hard to debug" was a bit of an overstatement I apologize. It's just that coming from a "mini-batch SGD DL" background, an ML training almost never takes less than a couple of minutes even for very very lightweight architectures and few data. Therefore I was not expecting such a low timeout. But I feel this is a fundamental difference between the theoretical optimization community that was the intended target audience of benchopt and the DL crowd. 

It does show up on the log, but since it does not fail or interrupt the flow of logs I did not pay attention and was careless and tried to find a bug in the code before I re-checked the logs to find that out. This is why I used the word "inattentive user".

Therefore my very biased viewpoint would be to have indeed a larger timeout.

Regardless of the value of the default passing a certain number of hours instead of seconds would be great !
Regarding the implementation details maybe pandas is indeed the way to go !


Thanks for the feedback anyways !
